src/Configuration/RectorConfigBuilder.php

```diff
@@ -246,6 +246,10 @@ public function __invoke(RectorConfig $rectorConfig): void
             $rectorConfig->sets($uniqueSets);
         }
 
+        // log rules from sets and compare them with explicit rules
+        $setRegisteredRectorClasses = $rectorConfig->getRectorClasses();
+        SimpleParameterProvider::addParameter(Option::SET_REGISTERED_RULES, $setRegisteredRectorClasses);
```

Comment:
this should be moved to `RectorConfig` sets method imo, so it will works on both `RectorConfig` and `RectorConfigBuilder`, since `RectorConfigBuilder` consume it.

Reply:
This should only work with `->withRules()` method, as more modern API. 
Adding it to `RectorConfig` would also break for sets, as the sets ulres `->rule()` method now.